### PR TITLE
Without db 008

### PR DESCRIPTION
**Pull request by mjkoo711**

### CODE REVIEW

File: back-end/app.py [hunk 1]
```diff
@@ -1,107 +1,113 @@
...
+from __future__ import print_function
+
+import json
+import os
+
+from flask import Flask, request, make_response
+
+import uuid
+import redis
+
```
Comment (shberry) [inline back-end/app.py]: remove line

File: back-end/app.py [hunk 2]
```diff
@@ -1,107 +1,113 @@
...
+import yelpCrawler
```
Comment (shberry) [inline back-end/app.py]: can we group the imports together?

File: back-end/app.py [hunk 3]
```diff
@@ -1,107 +1,113 @@
...
+    res = processWebhookRequest(req)
+
+    # JSON Encoding
+    res = json.dumps(res, indent=VAL_INDENT)
+
+    r = make_response(res)
+    r.headers['Content-Type'] = 'application/json'
+    return r
+
+
```
Comment (shberry) [inline back-end/app.py]: remove extra line

File: back-end/app.py [hunk 4]
```diff
@@ -1,107 +1,113 @@
...
+def callCrawler(req) :
+    result = req.get("result")
+    parameters = result.get("parameters")
+    parameter_atmosphere = parameters.get("atmosphere")
+    parameter_facility = parameters.get("mapsort")
+    parameter_open = parameters.get("open")
+
+    data = yelpCrawler.getYelpData(40.703491, -73.913351)
+    return data
+def makeWebhookResult(data):
```
Comment (shberry) [inline back-end/app.py]: add new line above?

File: back-end/app.py [hunk 5]
```diff
@@ -1,107 +1,113 @@
...
+    # Get json from dialogflow. 
+    req = request.get_json(silent=True, force=True)
+    res = processWebhookRequest(req)
+
+    # JSON Encoding
+    res = json.dumps(res, indent=VAL_INDENT)
+
+    r = make_response(res)
+    r.headers['Content-Type'] = 'application/json'
+    return r
```
Comment (shberry) [inline back-end/app.py]: can we user more meaningful variable name for 'r'?

File: back-end/yelpCrawler.py [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+def request(host, path, api_key, url_params=None):
+    """Given your API_KEY, send a GET request to the API.
+    Args:
+        host (str): The domain host of the API.
+        path (str): The path of the API after the domain.
+        API_KEY (str): Your API Key.
+        url_params (dict): An optional set of query parameters in the request.
+    Returns:
+        dict: The JSON response from the request.
+    Raises:
```
Comment (shberry) [inline back-end/yelpCrawler.py]: Not familiar with python world words, but is it normal to use 'Raises'? I'm more familiar with 'Throws' but up to you :)

File: back-end/yelpCrawler.py [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+            'Encountered HTTP error {0} on {1}:\n {2}\nAbort program.'.format(
+                error.code,
+                error.url,
+                error.read(),
+            )
+        )
+
+    # Merge loadJsonFile.py in this file
+    datainfo = businessList
+
```
Comment (shberry) [inline back-end/yelpCrawler.py]: maybe we can remove this line?

File: back-end/yelpCrawler.py [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+    datalist = []
+
+    for item in datainfo:
```
Comment (shberry) [inline back-end/yelpCrawler.py]: This for loop is quite big - can we add more comments within this for loop?

File: back-end/yelpCrawler.py [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+                if dataidx.name == "dd": 
+                    attr_content = dataidx.string.strip()
+            moreinfo_data[attr_name] = attr_content
+
+        data['attributes'] = moreinfo_data
+        datalist.append(data)
+
+    return json.dumps(datalist, indent=4)
+
+
```
Comment (shberry) [inline back-end/yelpCrawler.py]: remove extra line

**Review by shberry — COMMENTED** [review]
Cool!

**Review by khc196 — APPROVED** [review]
Cool!

**Pull request merged**